test/system/test_cvp_change_control_api.py

```diff
@@ -0,0 +1,784 @@
+STOP_NOTE = "Stop the CC via cvprac cc system tests"
+TASK_ID = None
+
+
+class TestCvpClientCC(TestCvpClientBase):
+    """ Test cases for the CvpClientCC class.
+    """
+    @classmethod
+    def setUpClass(cls):
+        """ Initialize variables
+        """
+        super(TestCvpClientCC, cls).setUpClass()
+
+    @classmethod
+    def tearDownClass(cls):
+        """ Reset variables value to None
+        """
+        super(TestCvpClientCC,
+              cls).tearDownClass()
+
+    def get_version(self):
+        """ Get api version
+        """
+        if self.clnt.apiversion is None:
+            self.api.get_cvp_info()
+        if self.clnt.apiversion > 3.0:
+            pprint(
+                'RUN TEST FOR V3 CHANGE CONTROL APIs')
+            return True
+        pprint(f'SKIPPING TEST FOR API - {self.clnt.apiversion}')
+        return False
+
+    def get_cc_status(self, cc_id):
+        """ Get change control status
+        """
+        params = f'key.id={cc_id}'
+        cc_url = f'/api/resources/changecontrol/v1/ChangeControl?{params}'
+        response = self.clnt.get(
+            cc_url, timeout=30)
+        return response
+
+    def get_device_list(self):
+        """ Get device list from inventory
+        """
+        inventory = self.api.get_inventory()
+        device_list = []
+        if len(inventory) >= 1:
+            # for i in range(0, len(inventory)):
+            #     device_list.append(
+            #         inventory[i]['serialNumber'])
+            device_list = [inventory[i]['serialNumber'] for i in range(len(inventory))]
+        else:
+            raise Exception("No device found")
+        return device_list
+
+    def create_snapshot(self):
+        """ Create snapshot for change control with custom stages
+        """
+        pprint('CREATING SNAPSHOT...')
+        device_list = self.get_device_list()
+        template_details = {
+            "commands": [
+                "show version"
+            ],
+            "deviceList": [
+                device_list[0]
+            ],
+            "frequency": "350",
+            "name": "show version"
+        }
+        response = self.clnt.post(
+            '/snapshot/templates/schedule?', data=template_details)
+        if response['status'] == "success":
+            template_id = response['templateKey']
+        else:
+            raise Exception(
+                "Snapshot not created")
+        return template_id
+
+    def delete_snapshot(self, snaps):
+        """ Delete snapshot
+        """
+        pprint('DELETING SNAPSHOT...')
+        response = self.clnt.delete(
+            '/snapshot/templates?', data=snaps)
+        return response
+
+    def create_task(self):
+        """ Create new task
+        """
+        pprint('CREATING TASKS...')
+        # global task_id
+        (task_id, _) = self._create_task()
+        self.task_id = task_id
+        return task_id
+
+    def create_change_control_for_tasks(self, task_id):
```

Comment:
Change this to create_change_control_for_task as this is setup to run for a single task_id and not a list of tasks.

Reply:
Updated!